src/durableorchestrationcontext.ts

```diff
@@ -202,8 +208,25 @@ export class DurableOrchestrationContext {
      * function.
      * @returns A Durable Task that completes when the called activity
      * function completes or fails.
+     *
+     * @deprecated this version of `callActivity()` is deprecated. Pass an object
+     * returned from `df.app.activity()` as the first argument instead.
      */
-    public callActivity(name: string, input?: unknown): Task {
+    public callActivity(name: string, input?: unknown): Task;
+
+    /**
+     * Schedules an activity function for execution
+     *
+     * @param activity The activity object returned from a `df.app.activity()` call.
+     * @param input The JSON-serializable input to pass to the activity
+     * function.
+     * @returns A Durable Task that completes when the called activity
+     * function completes or fails.
+     */
+    public callActivity(activity: CallActivityInput, input?: unknown): Task;

```

Comment:
Like call it just `Activity` and `Orchestration`?

Reply:
Yep, same for `Entity` I'd say

Reply:
I don't love that naming tbh, because it feels inaccurate, and also those are very special words in the SDK, we might need them later in more appropriate contexts. Besides, for `Entity`, it conflicts with the name of the class we already have. But since I have no better alternatives, I've made that change.

Reply:
> those are very special words in the SDK, we might need them later in more appropriate contexts

This concerns me as well. What about `RegisteredActivity` or `RegisteredActivityFunction`?

Reply:
I liked that. Updated to `RegisteredActivity`, `RegisteredOrchestration` and `RegisteredEntity`.